hpx/runtime/serialization/detail/non_default_constructible.hpp

```diff
@@ -0,0 +1,31 @@
+//  Copyright (c) 2017 Anton Bikineev
+//
+//  Distributed under the Boost Software License, Version 1.0.
+//  See accompanying file LICENSE_1_0.txt or copy at
+//  http://www.boost.org/LICENSE_1_0.txt)
+
+#ifndef HPX_SERIALIZATION_NON_DEFAULT_CONSTRUCTIBLE_HPP
+#define HPX_SERIALIZATION_NON_DEFAULT_CONSTRUCTIBLE_HPP
+
+#include <hpx/config.hpp>
+
+#include <memory>
+
+namespace hpx { namespace serialization { namespace detail
+{
+    ///////////////////////////////////////////////////////////////////////////
+    // default fall-backs for serializing non-default-constructible types
+    template <typename Archive, typename T>
+    void save_construct_data(Archive&, T const*, unsigned)
+    {
+    }
+
+    // by default fall back to in-place default construction
+    template <typename Archive, typename T>
+    void load_construct_data(Archive& ar, T* t, unsigned)
+    {
+        ::new (t) T;
```

Comment:
Hmm, I would assume, that you load all ctor arguments in that function that are needed to construct the object. After that, the object should be complete and should not require any additional serialization. I was under the impression that `load_construct_data` completely replaces the regular `serialize` function.